### ISSUE TITLE
Bulk Not Increasing with size

### ISSUE THREAD

**Issue by GoblinOfChaos** [foundryvtt/pf2e#7567]
The amount of bulk a creature can have as encumbered and max is not changing with size. Tiny, Small, Large all have base 5, max 10

**Comment by stwlam** [comment]
The values you are seeing are bulk for items appropriate to the size of the creature. See https://2e.aonprd.com/Rules.aspx?ID=256

**Comment by CarlosFdez** [comment]
To go into more detail, while the Bulk limit of a large creature is doubled (Bulk Limit column), the actual bulk of the items themselves is also reduced in the same tables. However, were we to implement both, then `Tiny or Large (or larger) creatures can usually wear and carry about the same amount of appropriately sized gear as a Medium creature.` would become an untrue statement. Therefore it is not RAI to implement all of it and we have to pick and choose.

Foundry chose to implement bulk item adjustments, but a different implementation that instead alters maximum bulk could be feasible. I think it'd be reasonable to implement a bulk methodology option that chooses between the two, but someone would have to be willing to implement it, and it'd need to be as uninvasive (aka clean) as possible because it turns out that most people don't actually care about encumbrance.

**Comment by TactiCool-HUN** [comment]
My main issue is with the decision on this is the following

some random maths (I ignored the overcharge stuff where foundry lets you carry 5.9 bulk on a 5bulk carry cap):
current foundry implementation (in extreme scenario):
carry cap 5
large pc
carry cap stays 5
9 bulk items are handled as light items
so a large pc can carry up to 450 bulk of "equivalent bulk" or idk how to call it

current foundry implementation (less extreme scenario):
carry cap 5
large pc
carry cap stays 5
some 2 bulk and some 1 bulk items mixed (half and half first, then fill out anything left with 1bulk)
all of those become light
so 50 items total, 25x 2bulk + 25x 1bulk = 75 of "equivalent bulk"

just reading 6-19 would say (extreme scenario):
carry cap 5
large pc
carry cap upgrades to 10
9 bulk is 9 bulk
can carry 1x 9bulk item
(if you wnat you can optionally fill it up with 10x 1 bulk (which goes as light) coming out to 19 "equivalent bulk" carried)

just reading 6-19 would say (less extreme):
carry cap 5
large pc
carry cap upgrades to 10
1 and 2 bulks equivally mixed then filled up with 1 bulk
2 bulks stay 2 bulk
1 bulks go to light
4x 2 bulk + 20x light bulk = 28 "equivalent bulk"

just reading 6-20 (extreme scenario)....
doesn't say anything about how bulks of lower size category items should be treated? just that if you make an item for of that size category than it changes in weight/price?
like where is the contradiction here?

6-20 for example says:
if you have a large pc, and want to make it an item that is by default light bulk, than it becomes 1 bulk
which the large pc due to 6-19 treats as light
if you want to make it an item that is 2 bulk by default, than the large sized version is 4 bulk
which the large pc counts as 4 bulk, but since they have double the carry cap it will still end up "feeling like 2 bulks" for them

Paizo's system has zero contradiction here, at all, it just simply works, and I'm sad I have to fiddle around with items to make it actually work in foundry :c

**Comment by stwlam** [comment]
> some random maths (I ignored the overcharge stuff where foundry lets you carry 5.9 bulk on a 5bulk carry cap)

You may want to start with some basics: https://2e.aonprd.com/Rules.aspx?ID=2155

> Ten light items count as 1 Bulk, and you round down fractions (so 9 light items count as 0 Bulk, and 11 light items count as 1 Bulk).

**Comment by TactiCool-HUN** [comment]
I'm so glad that you picked out the like 1 part that is negligable in the final results of this

**Comment by CarlosFdez** [comment]
Equivalent bulk refers to a large creature carrying large items vs a medium creature carrying medium items. Results should be roughly the same. It does not refer to a creature carrying incorrect sized items. Once items are mismatched there is no longer any expectation of how much beyond large creature carries more medium items than medium creature. This is just the natural consequence of the 1 bulk becomes 1L that paizo published.

**Comment by CarlosFdez** [comment]
Also, the alternative is to double a large PCs carrying capacity, but then make a large dagger 1 bulk instead of 1L (10x). An enlarged PC with a bunch of daggers would be immediately encumbered.

**Comment by TactiCool-HUN** [comment]
Someone shed some light on why I found this way more extreme than it should be: Foundry makes 1 to 9 bulk equal 1L (and 10 to 19 bulk as 1 bulk), while it would probably make much more sense to make x bulk count as xL and let it just scale by that. (dunno if it's feasable to do)

Thank you for showing me what the problem is with the official system, which, I do agree isn't good.

**Personally** the only thing I'd change from Paizo's official system is probably take out that huge step up instead of introducing a more agressive step down with different size items (making L go to 2L if you are large; granting them the double carry cap; and keeping that 2 bulk is 2 bulk even if you are large). But at this point it's probably personal preference.

**Comment by CarlosFdez** [comment]
I have to double check what's going on, because last I checked a medium item on a large creature was supposed to halve, and weird behavior was only supposed to happen once it went past the 1 bulk threshold. So 9 bulk should become 4 bulk. Weird.

I don't like the official system implementation either. There's a lot of other implementations I'd prefer as a player, including halving 1 bulk to 5L instead of 1L or adjusting the bulk limit and the item sizes in the absolute (large creatures carry 2x, large longswords 2 bulk regardless of creature size), but I'm torn between implementing what I want and implementing what they wrote.

**Comment by TactiCool-HUN** [comment]
> Also, the alternative is to double a large PCs carrying capacity, but then make a large dagger 1 bulk instead of 1L (10x). An enlarged PC with a bunch of daggers would be immediately encumbered.

@CarlosFdez 

actually just realized, not true, as L bulk daggers become 1 bulk when enlarged, yes. However! Large PCs treat 1 bulk as L bulk based on 6-19. So for them personally nothing changes.

I made a little google sheets that I think is completely RAW (based on 6-19 and 6-20, do let me know if there is another rule section that you think plays into it). It will actually math out to just be the exact same, as someone above mentioned how it should feel the same.

https://docs.google.com/spreadsheets/d/1vf0n8fKsbeOea0Z_f1JTLqGuGBEw3r6SLYvz-9tZneo/edit?usp=sharing

**Comment by Manypiemen12** [comment]
I just wanted to jump in here because I'm having my own issues with the system. Namely, with horses and players mounting them. I'm using the ridable module (Which is just fantastic by the way) and we just got some mounting weight things fixed, and it annoys me that weight is halved, rounding down. If a PC is a total weight of 9, it should count as 4.5 bulk on the horse, which matters if 2 such players are riding the same horse that 1 bulk makes a difference.

I agree, no one like tracking encumbrance rules. It's a pain. But I feel that is largely the point of a system like Foundry. So it can track it for you. You throw the items on the player or horse, and it tells you if it's too much. Takes a lot of the math off the average player's shoulder's, which is why it's important that the painful leg work is done now rather than make yet another DM look at the system and think "Why is it doing it this way?"

**Comment by Manypiemen12** [comment]
With my previous comment in mind, would anyone here know what function inside the module handles the halfling for large creatures? I would rather not have to go looking for it, but until such time that it gets "fixed" i would like to manually adjust the halfling to include the half bulks, if I can.

**Comment by CarlosFdez** [comment]
@TactiCool-HUN I saw that in a previous release the logic had changed (it was trying to fix some issue with armor), so its back to it being mostly symmetrical, so smaller items halve on larger pcs rather than 1/10. And the armor scaling looks to be the same. 1/2 instead of 1/10 satisfies a lot of those same tables in the listed examples, and should behave more sanely.

@Manypiemen12 the thing that was being brought up is that in that scenario with the horse, it was  1/10 instead of 1/2. It has since been changed to 1/2 and there is nothing to fix. It looks like most of your issue is with the rules themselves (I don't like the bulk rules either), and what you want is the implementation of homebrew bulk rules, which is outside of the scope of the system. You should probably request or commission a module.

Specifically, L bulk is not a countable unit. 4.5 bulk nor 4 bulk 5L isn't a valid value. We show 5L for display purposes but it doesn't actually exist. The tables also make every effort to round.

**Comment by Manypiemen12** [comment]
If I'm missing a rules example somewhere, please point it out, but as far as I know, there is nowhere is PF2E that deals with halving bulk. That is to say, that this implementation of the Bulk of large creatures is causing a problem that the system never accounted for.

I agree that half a bulk is not a thing the system accounts for, but it was never designed to account for that. We are creating the problem of half bulks by halving the bulks for Large creatures instead of just doubling their carry capacity, as written, as then sweeping that problem under the rug by saying half bulks aren't a thing.

As it currently stands, if I put 6 items of 3 bulk each on a horse, or carried by a large player, I am getting 12 free bulk space of carry capacity. It's definitely a less common occurrence, but that is an egregious oversight, in my opinion.

Basically, what I'm asking for is that if we are going to implement Large creatures in this manner, let's at least patch out the issues with that implementation. I guess I'm not fully understanding what the reluctance to do that would be? I am really just at the start of my Foundry programming knowledge, but if it's just a matter of man power, I'm willing to do the work to implement something like that. I just want to understand why we wouldn't want to do that.

**Comment by CarlosFdez** [comment]
Its not an issue of manpower, its an issue of system goals. We're trying to implement bulk as close to RAW as possible, not as well designed as possible.

There are a few contradicting elements in the rules:
- Creatures treat different sized items as different bulk: https://2e.aonprd.com/Rules.aspx?ID=2165
- Creatures of different sizes treat certain different sized items as light/negligible bulk but also have a bulk limit change: https://2e.aonprd.com/Rules.aspx?ID=2164
- `Because the way that a creature treats Bulk and the Bulk of gear sized for it scale the same way, Tiny or Large (or larger) creatures can usually wear and carry about the same amount of gear as a Medium creature.`

Because of multiple references to treating thing as a different size, and the phrase that they can carry the same amount of appropriately sized gear, we have to read that the sizes of things is what gets adjusted. If we also change bulk limit, large creatures end up carrying more since there is no bulk adjustment for equal sized items, which is in direct contradiction.

Paizo would either have to errata this system with a better designed one, or it can only be in a module.

**Comment by Manypiemen12** [comment]
SO essentially with this implementation, the idea is that, for a large creature, if you have an appropriately sized item, you scale the item up(- to L to 1 to x2) but then because they are Large, you scale the item back down, and it cancels out?

That does make sense to me, from that stance. I think the biggest issue is then when dealing with items that did not get scaled. That's where it breaks down, and I guess I really can't tell for sure what Paizo's idea was here.

Sounds like I need to get started on my own Module then. Or piggy back off one of the homebrew ones.
